src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -273,15 +332,35 @@ public GroupPrincipal pushGroupDefinitions(
     }
     journal.recordGroupPushStarted();
     String gsaVerString = config.getGsaVersion();
-    if (!new GsaVersion(gsaVerString).isAtLeast("7.2.0-0")) {
+    GsaVersion gsaVersion = new GsaVersion(gsaVerString);
+    if (!gsaVersion.isAtLeast("7.2.0-0")) {
       log.log(Level.WARNING,
-          "GSA ver {0} doesn't accept group definitions", gsaVerString);
+          "GSA ver {0} does not accept group definitions", gsaVerString);
       journal.recordGroupPushFailed();
       return defs.entrySet().iterator().next().getKey();
     }
+    if (feedType == FULL && !gsaVersion.isAtLeast("7.4.0-0")) {
+      log.log(Level.WARNING,
+          "GSA ver {0} does not support per-source replacement of all groups",
+          gsaVerString);
+      feedType = INCREMENTAL;
+    }
     if (null == handler) {
       handler = defaultErrorHandler;
     }
+    if (groupSource == null) {
+      groupSource = config.getFeedName();
+    }
+    log.log(Level.INFO, "Starting {0} push of {0} groups from source {1}",
+        new Object[] { feedType, defs.size(), groupSource });
+    String feedSourceName;
+    if (feedType == INCREMENTAL) {
+      feedSourceName = previousGroupSource(groupSource);
+    } else {
+      feedSourceName = alternateGroupSource(groupSource);
```

Comment:
I would think that ACL resolution on the GSA would be the same as if you had multiple connector instances, each feeding groups using a different feed.name.